### ISSUE TITLE
Investigate transitive dependencies in 1.0.6-SNAPSHOT

### ISSUE THREAD

**Issue by saturnism** [GoogleCloudPlatform/cloud-sql-jdbc-socket-factory#65]
In #64, the Socket Factory now pulls in Jetty Unix Socket dependency, which has quite a few transitive dependencies.

Do we need them? I'm expecting Servlet API and Jetty HTTP Server to be unnecessary. The PR also only uses `jnr` and nothing from Jetty.

This version should not be published/released until the dependencies are sorted out.

```
[INFO] \- com.google.cloud.sql:mysql-socket-factory:jar:1.0.6-SNAPSHOT:compile
[INFO]    +- mysql:mysql-connector-java:jar:5.1.45:compile
[INFO]    +- com.google.cloud.sql:jdbc-socket-factory-core:jar:1.0.6-SNAPSHOT:compile
[INFO]    |  \- com.google.apis:google-api-services-sqladmin:jar:v1beta4-rev25-1.22.0:compile
[INFO]    |     \- com.google.api-client:google-api-client:jar:1.22.0:compile
[INFO]    |        +- com.google.oauth-client:google-oauth-client:jar:1.22.0:compile
[INFO]    |        |  +- com.google.http-client:google-http-client:jar:1.22.0:compile
[INFO]    |        |  |  \- org.apache.httpcomponents:httpclient:jar:4.0.1:compile
[INFO]    |        |  |     +- org.apache.httpcomponents:httpcore:jar:4.0.1:compile
[INFO]    |        |  |     +- commons-logging:commons-logging:jar:1.1.1:compile
[INFO]    |        |  |     \- commons-codec:commons-codec:jar:1.3:compile
[INFO]    |        |  \- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO]    |        +- com.google.http-client:google-http-client-jackson2:jar:1.22.0:compile
[INFO]    |        |  \- com.fasterxml.jackson.core:jackson-core:jar:2.1.3:compile
[INFO]    |        \- com.google.guava:guava-jdk5:jar:17.0:compile
[INFO]    \- org.eclipse.jetty:jetty-unixsocket:jar:9.4.8.v20171121:compile
[INFO]       +- org.eclipse.jetty:jetty-server:jar:9.4.8.v20171121:compile
[INFO]       |  +- javax.servlet:javax.servlet-api:jar:3.1.0:compile
[INFO]       |  +- org.eclipse.jetty:jetty-http:jar:9.4.8.v20171121:compile
[INFO]       |  |  \- org.eclipse.jetty:jetty-util:jar:9.4.8.v20171121:compile
[INFO]       |  \- org.eclipse.jetty:jetty-io:jar:9.4.8.v20171121:compile
[INFO]       \- com.github.jnr:jnr-unixsocket:jar:0.18:compile
[INFO]          +- com.github.jnr:jnr-ffi:jar:2.1.4:compile
[INFO]          |  +- com.github.jnr:jffi:jar:1.2.15:compile
[INFO]          |  +- com.github.jnr:jffi:jar:native:1.2.15:runtime
[INFO]          |  +- org.ow2.asm:asm:jar:5.0.3:compile
[INFO]          |  +- org.ow2.asm:asm-commons:jar:5.0.3:compile
[INFO]          |  +- org.ow2.asm:asm-analysis:jar:5.0.3:compile
[INFO]          |  +- org.ow2.asm:asm-tree:jar:5.0.3:compile
[INFO]          |  +- org.ow2.asm:asm-util:jar:5.0.3:compile
[INFO]          |  \- com.github.jnr:jnr-x86asm:jar:1.0.2:compile
[INFO]          +- com.github.jnr:jnr-constants:jar:0.9.8:compile
[INFO]          +- com.github.jnr:jnr-enxio:jar:0.16:compile
[INFO]          \- com.github.jnr:jnr-posix:jar:3.0.35:compile
```

**Comment by saturnism** [comment]
I only see uses `jnr-unixsocket`. I don't see any uses of Jetty related components.

Once we figure out what's needed, we need to review what are exposed to the user:
- Since this is a socket factory, we shouldn't care pull in `mysql-connector-java`. That should be brought in by the library user.  This should probably be scoped to `provided`.
- `guava-jdk5` will also break others who are using regular `guava`. In fact, `google-cloud-java` uses regular `guava`. This will not work well together.  This should be removed and replaced w/ regular `guava`.
- Optionally, shade as needed.

**Comment by kurtisvg** [comment]
This issue was resolved in PR #71 and was released in v1.0.8.
